apps/projects/routes.py

```diff
@@ -1,14 +1,38 @@
+import base64
+import logging
+import os
+import re
+
 from datetime import datetime
 
 from bson import json_util
-from flask import current_app as app, request, Response
+from flask import abort, request, make_response
+from flask import current_app as app
 from flask.views import MethodView
 
-from lib.video_editor import get_video_editor
-from lib.utils import create_file_name, format_id
-from lib.errors import bad_request, not_found, forbidden
+from lib.errors import bad_request, forbidden, not_found
+from lib.utils import create_file_name, format_id, json_response, represents_int
 from lib.validator import Validator
+from lib.video_editor import get_video_editor
 from . import bp
+from .tasks import task_get_list_thumbnails, task_edit_video
+
+logger = logging.getLogger(__name__)
+
+
+def check_user_agent():
+    user_agent = request.headers.environ.get('HTTP_USER_AGENT')
+
+    client_name = user_agent.split('/')[0]
+    if client_name.lower() not in app.config.get('AGENT_ALLOW'):
+        abort(bad_request("client is not allow to edit"))
+    return user_agent
+
+
+def check_request_schema_validity(request_schema, schema):
```

Comment:
It might be useful to return a `validator.document` and use this `clean` data if you need it
https://docs.python-cerberus.org/en/stable/usage.html#fetching-processed-documents